test-loop-tests/src/examples/delayed_receipts.rs

```diff
@@ -0,0 +1,116 @@
+use std::iter::repeat_with;
+
+use itertools::Itertools;
+use near_async::time::Duration;
+use near_o11y::testonly::init_test_logger;
+use near_primitives::gas::Gas;
+use near_primitives::receipt::{DelayedReceiptIndices, StateStoredReceipt};
+use near_primitives::test_utils::create_user_test_signer;
+use near_primitives::transaction::{ExecutionStatus, SignedTransaction};
+use near_primitives::trie_key::TrieKey;
+use near_primitives::types::{Balance, Nonce, ShardId};
+
+use crate::setup::builder::TestLoopBuilder;
+use crate::utils::account::{
+    create_account_id, create_validators_spec, validators_spec_clients_with_rpc,
+};
+use crate::utils::node::TestLoopNode;
+
+/// Example test that creates a chunk which, when applied, creates a delayed receipt.
+/// Requires "test_features" feature to be enabled.
+#[test]
+fn delayed_receipt_example_test() {
+    init_test_logger();
+
+    let user_account = create_account_id("user");
+    let validators_spec = create_validators_spec(1, 0);
+    let clients = validators_spec_clients_with_rpc(&validators_spec);
+    let gas_limit = Gas::from_teragas(300);
+    let genesis = TestLoopBuilder::new_genesis_builder()
+        .shard_layout_single_shard()
+        .validators_spec(validators_spec)
+        .gas_limit(gas_limit)
+        .add_user_account_simple(user_account.clone(), Balance::from_near(10))
+        .build();
+    let mut env = TestLoopBuilder::new()
+        .genesis(genesis)
+        .epoch_config_store_from_genesis()
+        .clients(clients)
+        .build()
+        .warmup();
+
+    let rpc_node = TestLoopNode::rpc(&env.node_datas);
+    let mut nonce: Nonce = 0;
+    let mut next_nonce = || {
+        nonce += 1;
+        nonce
+    };
+
+    let deploy_test_contract_tx = SignedTransaction::deploy_contract(
+        next_nonce(),
+        &user_account,
+        near_test_contracts::rs_contract().to_vec(),
+        &create_user_test_signer(&user_account),
+        rpc_node.head(env.test_loop_data()).last_block_hash,
+    );
+    rpc_node.run_tx(&mut env.test_loop, deploy_test_contract_tx, Duration::seconds(2));
+
+    // Each transaction generates local receipt consuming more than a half
+    // the chunk space, so chunk can only fit 2 such receipts.
+    let gas_to_burn = gas_limit.checked_div(2).unwrap().checked_add(Gas::from_gas(1)).unwrap();
+    let txs = repeat_with(|| {
+        SignedTransaction::call(
+            next_nonce(),
+            user_account.clone(),
+            user_account.clone(),
+            &create_user_test_signer(&user_account),
+            Balance::ZERO,
+            "burn_gas_raw".to_owned(),
+            gas_to_burn.as_gas().to_le_bytes().to_vec(),
+            gas_limit,
+            rpc_node.head(env.test_loop_data()).last_block_hash,
+        )
```

Comment:
As an idea not related to this PR it might make sense to make it easier to create txs with rpc nodes (since likely rpc nodes are the ones usually used for sending txs). In that api deriving user signer, nonce and latest block hash can be automatic.